numpy/random/bit_generator.pyi

```diff
@@ -0,0 +1,126 @@
+import abc
+from threading import Lock
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    List,
+    Literal,
+    NamedTuple,
+    Optional,
+    Sequence,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+    overload,
+)
+
+from numpy import dtype, ndarray, uint32, uint64, unsignedinteger
+from numpy.typing import (
+    DTypeLike,
+    _ArrayLikeInt_co,
+    _DTypeLikeUInt,
+    _ShapeLike,
+    _SupportsDType,
+)
+
+_T = TypeVar("_T")
+
+_UIntType = TypeVar("_UIntType", uint64, uint32)
+_DTypeLike = Union[
+    Type[_UIntType],
+    dtype[_UIntType],
+    _SupportsDType[dtype[_UIntType]],
+]
+
+class _Interface(NamedTuple):
+    state_address: Any
+    state: Any
+    next_uint64: Any
+    next_uint32: Any
+    next_double: Any
+    bit_generator: Any
+
+class ISeedSequence(abc.ABC):
+    @overload
+    @abc.abstractmethod
+    def generate_state(
+        self, n_words: int, dtype: _DTypeLike[_UIntType] = ...
+    ) -> ndarray[Any, dtype[_UIntType]]: ...
+    @overload
+    @abc.abstractmethod
+    def generate_state(
+        self, n_words: int, dtype: _DTypeLikeUInt = ...
+    ) -> ndarray[Any, dtype[unsignedinteger[Any]]]: ...
+
+class ISpawnableSeedSequence(ISeedSequence):
+    @abc.abstractmethod
+    def spawn(self: _T, n_children: int) -> List[_T]: ...
+
+class SeedlessSeedSequence(ISpawnableSeedSequence):
+    @overload
+    def generate_state(
+        self, n_words: int, dtype: _DTypeLike[_UIntType] = ...
+    ) -> ndarray[Any, dtype[_UIntType]]: ...
+    @overload
+    def generate_state(
+        self, n_words: int, dtype: _DTypeLikeUInt = ...
+    ) -> ndarray[Any, dtype[unsignedinteger[Any]]]: ...
+    def spawn(self: _T, n_children: int) -> List[_T]: ...
+
+class SeedSequence(ISpawnableSeedSequence):
+    entropy: Union[None, int, Sequence[int]]
+    spawn_key: Tuple[int, ...]
+    pool_size: int
+    n_children_spawned: int
+    pool: ndarray[Any, dtype[uint32]]
+    def __init__(
+        self,
+        entropy: Union[None, int, Sequence[int]] = ...,
+        *,
+        spawn_key: Tuple[int, ...] = ...,
```

Comment:
```suggestion
        spawn_key: Sequence[int] = ...,
```

Reply:
Fixed, thanks.

Reply:
`spawn_key` is still annotated as `Tuple[int, ...]` here.

Reply:
It is always a `Tuple[int,,...]`  on the attribute, even though externally `Sequence[int]` is accepted.

Reply:
Yes, and I'm specifically talking about the function-argument here, _not_ the attribute. 

Requiring the function-argument to passed as a tuple is overly strict if the constructor itself explicitly performs a `tuple()` call.

Reply:
Grrr.  You are right.  I had them backward and somehow couldn't copy the differences.

Reply:
Fixed now.